runtimes/runtimes/base-runtime.ts

```diff
@@ -116,8 +116,10 @@ export const baseRuntime = (connections: { reader: MessageReader; writer: Messag
             copyFile: (_src, _dest, _options?) => Promise.resolve(),
             exists: _path => Promise.resolve(false),
             getFileSize: _path => Promise.resolve({ size: 0 }),
-            getServerDataDirPath: _serverName => '',
+            getServerDataDirPath: serverName =>
+                lspRouter.clientInitializeParams!.initializationOptions?.aws?.clientDataFolder ?? `/${serverName}`,
             getTempDirPath: () => '/tmp',
+            getUserHomeDir: () => '',
```

Comment:
if "/tmp" is a valid default on web, maybe "/home/user" is a valid default too :) 

does `runtimes/webworker.ts` need to auto-create those directories somehow? (probably out of scope for this PR, but just curious)

Reply:
Thats a good question. What would creating those directories look like in a web worker since there isn't an actual file system? Is there other other initialization necessary?

Reply:
Not sure about web *workers*, but web (html) has a filesystem api (isolated per browser-tab): 
https://developer.mozilla.org/en-US/docs/Web/API/FileSystem

I would guess that api (or something like it) is used by web apps like http://github.dev/ for all of the files/directories the user interacts with, including in the terminal/shell that runs in vscode-web.